src/linker/Linker/LinkContext.cs

```diff
@@ -418,15 +418,15 @@ public bool IsOptimizationEnabled (CodeOptimizations optimization)
 			return (DisabledOptimizations & optimization) == 0;
 		}
 
-		public void LogMessage (string message, params object[] values)
```

Comment:
> Removing this property setter ("rename to SetLogger()") would thus result in a compatibility change

Speaking of "compatibility changes", this commit *broke* xamarin-android compilation, as it removed the `LinkContext.LogMessage(string, params object[])` overload, resulting in CS1501 errors in xamarin-android:

https://github.com/xamarin/xamarin-android/commit/856460dbe2ec65cb5ca1111880ab8940419fe945

This raises a somewhat important question: *what are* the compatibility constraints on this repo?  Is source compatibility supposed to be present?  Binary compatibility?  No compatibility?

If compatibility is required, this repo may want to look into using `Microsoft.DotNet.ApiCompat.exe`.  See also?

  * https://github.com/xamarin/xamarin-android/commit/07e7477126fdc3eb68cca38347d1b8698d334874
  * https://github.com/xamarin/xamarin-android/blob/master/tests/api-compatibility/README.md

Reply:
For existing Xamarin like workloads, there is no compatibility as we integrate at the source code level. Going forward with .NET 5 the plan is to switch to binary backward compatibility.